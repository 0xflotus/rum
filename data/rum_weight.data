As a reward for your reformation I write to you on this precious sheet.|write
You see I have come to be wonderfully attached to Heidelberg, the|attached come see
beautiful, the quaint, the historically poetic, learned and picturesque|
old town on the Neckar. It seems like another home. So I could not show|seems show could
my appreciation of you in a more complimentary way than by sending this|sending
little series of pictures. Have you ever been here, I wonder? You did|did have been wonder
not say, but you wrote as if you knew it by sight as well as by heart.|wrote say knew
As I cannot know, I will venture an explanation. The panorama speaks for|know will speaks
itself. Put on your "specs" and look at the castle, half way up the|put look
_berg_, "the Jettenhuhl, a wooded spur of the Konigestuhl." Look at it|Look
from the "Terrasse." Thus you'll get something of an idea of it. The|get
Gesprente Thurm is the one that was blown up by the French. The|is blown was
thickness of the walls, twenty-one feet, and the solid masonry, held it|held
so well that only a fragment, as it were, gave way. It still hangs as if|were gave hangs
ready to be replaced. "Das Grosse Fass Gebaude," too, you will have no|be replaced will have
difficulty in making out. If you only had it with its 49,000 gallons of|making had
wine, but wouldn't you divide with your neighbors! The columns in the|wouldn't divide
portico that shows in the Schlosshof are the four brought from|shows are brought
Charlemagne's palace at Ingelheim by the Count Palatine Ludwig, some|
time between 1508-44. The Zum Ritter has nothing to do with the castle,|has do
but is an ancient structure (1592) in the Renaissance style, and one of|is
the few that escaped destruction in 1693. It is a beautiful, highly|escaped is
ornamental building, and I wish you could see it, if you have not seen|wish could see have seen
it.|
|
All the above information, I beg you to believe, I do not intend you|beg believe do intend
to think was evolved from my inner consciousness, but gathered from|think was evolved gathered
the--nearest guide-book!|
|
I am so much obliged to you for mapping out Switzerland to me. I have|am obliged have
been trying my best to get all those "passes" into my brain. Now, thanks|been trying get
to your letter, I have them all in the handiest kind of a bunch. Ariel|have
like, "I'll do my bidding gently," and as surely, if I get there. But|do bidding get
there are dreadful reports of floods and roads caved in and bridges|are caved
swept away and snows and--enough of such exciting items as sets one|swept sets
thinking--"to go or not to go?" We are this far on the way. Reached|thinking go go are Reached
here this afternoon. Have spent the evening sauntering in the gardens,|Have spent sauntering
the Conversationhaus, the bazaar, mingling with the throng, listening to|mingling listening
the band, and comparing what it is with what it was. It was a gay and|comparing was was
curious spectacle, but on the whole had "the banquet-hall deserted"|had deserted
look. The situation is most beautiful. It lies, you know, at the|is lies know
entrance of the Black Forest, among picturesque, thickly-wooded hills,|
in the valley of the Oos, and extends up the slope of some of the hills.|extends
The Oos is a most turbid, turbulent stream; dashes through part of the|is
town with angry, headlong speed. There is an avenue along its bank of|is
oaks, limes and maples, bordered with flower-beds and shrubberies, and|
adorned with fountains and handsome villas. We shall devote to-morrow to|
seeing all there is to be seen, and go to Strassburg to-morrow evening|seeing is be seen go
for two or three days. From there to Constance, and then hold _our_|
"Council" as to further movements.|
def fgr|
def xxx fgr|
