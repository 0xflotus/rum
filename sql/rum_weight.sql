CREATE TABLE testweight_rum( t text, a tsvector, r text );

CREATE FUNCTION fill_weight_trigger() RETURNS trigger AS $$
begin
  new.a :=
     setweight(to_tsvector('pg_catalog.english', coalesce(new.r,'')), 'A') ||
     setweight(to_tsvector('pg_catalog.english', coalesce(new.t,'')), 'D');
  return new;
end
$$ LANGUAGE plpgsql;

CREATE TRIGGER tsvectorweightupdate
BEFORE INSERT OR UPDATE ON testweight_rum
FOR EACH ROW EXECUTE PROCEDURE fill_weight_trigger();

CREATE INDEX rumidx_weight ON testweight_rum USING rum (a rum_tsvector_ops);

\copy testweight_rum(t,r) from 'data/rum_weight.data' DELIMITER '|' ;

SET enable_seqscan=off;
SET enable_indexscan=off;

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'ever:A|wrote');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'have:A&wish:DAC');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'have:A&wish:DAC');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'among:ABC');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'structure:D&ancient:BCD');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(complimentary:DC|sight)&(sending:ABC|heart)');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '!gave:D & way');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(go<->go:a)&(think:d<->go)');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(go<->go:a)&(think:d<2>go)');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & (!reach:a | way<->reach)');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & (!reach:a & way<->reach)');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & go & !way:a');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'show:d & seem & !town:a');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '!way:a');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & !way:a');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & !way:a');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & go');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'think<->go:d | go<->see');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d<->think');
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach<->think');


