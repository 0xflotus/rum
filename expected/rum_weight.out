CREATE TABLE testweight_rum( t text, a tsvector, r text );
CREATE FUNCTION fill_weight_trigger() RETURNS trigger AS $$
begin
  new.a :=
     setweight(to_tsvector('pg_catalog.english', coalesce(new.r,'')), 'A') ||
     setweight(to_tsvector('pg_catalog.english', coalesce(new.t,'')), 'D');
  return new;
end
$$ LANGUAGE plpgsql;
CREATE TRIGGER tsvectorweightupdate
BEFORE INSERT OR UPDATE ON testweight_rum
FOR EACH ROW EXECUTE PROCEDURE fill_weight_trigger();
CREATE INDEX rumidx_weight ON testweight_rum USING rum (a rum_tsvector_ops);
\copy testweight_rum(t,r) from 'data/rum_weight.data' DELIMITER '|' ;
SET enable_seqscan=off;
SET enable_indexscan=off;
SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'ever:A|wrote');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'have:A&wish:DAC');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'have:A&wish:DAC');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'among:ABC');
 count 
-------
     0
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'structure:D&ancient:BCD');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(complimentary:DC|sight)&(sending:ABC|heart)');
 count 
-------
     2
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '!gave:D & way');
 count 
-------
     3
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(go<->go:a)&(think:d<->go)');
 count 
-------
     0
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '(go<->go:a)&(think:d<2>go)');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & (!reach:a | way<->reach)');
 count 
-------
     2
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & (!reach:a & way<->reach)');
 count 
-------
     0
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & go & !way:a');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'show:d & seem & !town:a');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', '!way:a');
 count 
-------
    52
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'go & !way:a');
 count 
-------
     2
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & !way:a');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d & go');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'think<->go:d | go<->see');
 count 
-------
     1
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach:d<->think');
 count 
-------
     0
(1 row)

SELECT count(*) FROM testweight_rum WHERE a @@ to_tsquery('pg_catalog.english', 'reach<->think');
 count 
-------
     1
(1 row)

